.github/workflows/clang.yml

```diff
@@ -0,0 +1,133 @@
+name: Build clang
+# idea - only run on tagged builds?
+# also: opam installs use versions from infer v0.17.0
+# https://github.com/facebook/infer/blob/v0.17.0/opam.locked
+on: [push, pull_request]
+jobs:
+  opam:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+    env:
+      OCAML_VERSION: ocaml-variants.4.07.1+flambda
+      OPAM_VERSION: 2.0.3
+      OPAMYES: 1
+      OPAMVERBOSE: 1
+    steps:
+      - name: Install macOS build requirements
+        if: runner.os == 'macOs'
+        run: brew install pkg-config
+
+      - name: Set up opam
```

Comment:
I think we can skip all this opam stuff if the goal is only to build clang.

Reply:
Just to clarify: doesn't `infer` require a version of clang that is built with ocaml bindings? When clang is built, it'll only enable ocaml bindings if ocaml's `ctypes` and `ounit` (at least) are found.

Reply:
I see, I thought this was for clang-ocaml. No, infer doesn't need the OCaml bindings of clang.

(The way we interface with clang is that this repo defines a C++ plugin for clang that spits out the AST in some serialised format that is read on the OCaml side.)